### PR TITLE
ci: add a spam comment detection

### PR DESCRIPTION
**Pull request by Keith-CY**
When a comment is added in an issue

1. if the comment has specific words, it will be deleted
2. if the comment has unexpected links, an annotation about safety will be appended.

### CODE REVIEW

**Review by yanguoyu — APPROVED** [review]

File: .github/workflows/spam-comment-detection.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    steps:
+      - name: Set up Node.js
+        uses: actions/setup-node@main
+
+      - name: Check for Spam
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const comment = process.env.COMMENT_BODY
+            const spam_words = ['Neuron Support Request']
```
Comment (yanguoyu) [inline .github/workflows/spam-comment-detection.yaml:19]: Could we add this variable to GitHub action env?

File: .github/workflows/spam-comment-detection.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            const comment_id = process.env.COMMENT_ID
+            const issue_number = process.env.ISSUE_NUMBER
+            const owner = process.env.REPO_OWNER
+            const repo = process.env.REPO_NAME
+            const EXTERNAL_LINK_REGEXT = /https:\/\/(?!((\w+\.)?github\.com|github\.com|(\w+\.)?magickbase\.com|(\w+\.)?nervos\.org))/gi
+            if (spam_words.some(w => comment.includes(w))) {
+              console.info(`Spam comment: ${comment}`)
+              github.rest.issues.deleteComment({ owner, repo, comment_id })
+            } else if (EXTERNAL_LINK_REGEXT.test(comment)) {
```
Comment (yanguoyu) [inline .github/workflows/spam-comment-detection.yaml:28]: Should we skip the verification when the creator is a collaborator?

File: .github/workflows/spam-comment-detection.yaml (hunk 1 above)
Comment (Keith-CY) [reply .github/workflows/spam-comment-detection.yaml]: Update by loading spam word list from the secrets

https://github.com/nervosnetwork/neuron/pull/3288/files#diff-c369c155176306e9be1a7df1156a1408083a4b28c9bc392758336e0ec9ff8e05R18

File: .github/workflows/spam-comment-detection.yaml (hunk 2 above)
Comment (Keith-CY) [reply .github/workflows/spam-comment-detection.yaml:28]: I think this annotation should be for all links and users should be cautious even if the message is from a collaborator.

**Comment by Keith-CY** [comment]
Please have a review @homura @devchenyan

**Review by homura — APPROVED** [review]

**Pull request merged**